lib/Common/Memory/SmallFinalizableHeapBucket.h

```diff
@@ -174,7 +173,7 @@ class HeapBucketGroup
         typedef typename SmallHeapBlockType<objectAttributes, TBlockAttributes>::BucketType BucketType;
         static BucketType& GetBucket(HeapBucketGroup<TBlockAttributes> * heapBucketGroup)
         {
-            CompileAssert(objectAttributes & FinalizeBit);
+            Assert(objectAttributes & FinalizeBit);
```

Comment:
because the compiler is not smart enough, that while allocating with LeafBit, we already pick the Leaf buckets via the specialization, and should not fall through here. see Recycler.inl line 97 for detail.

---
In reply to: [89901040](https://github.com/Microsoft/ChakraCore/pull/2105#discussion_r89901040) [](ancestors = 89901040)